### ISSUE TITLE
Implement checks on test plan level

### ISSUE THREAD

**Issue by thrix** [teemtee/tmt#2676]
Seems we have use cases also for checks on plan level. For example users would like to sosreport after the plan run.

Crosslink: https://issues.redhat.com/browse/TFT-1469

**Comment by lukaszachy** [comment]
I have naming issue with having 'sosreport' executed as 'check'.. That tool just harvests logs, doesn't have capability to have an opinion about them. 

But +1 to have common way to gather logs, not only after tests but even before. Usually I need  to know which rpm are installed and having one 'rpm -qa' would be nicer that checking logs from prepare (not to mention that package might be not required explicitly and be preinstalled in VM/machine...)

While sosreport works for RH-based distros we might also need more generic way to gather final dmesg/journal/logs

**Comment by psss** [comment]
Seems there is an easy workaround for this:

```yaml
finish:
    how: shell
    script: dnf install -y sos && sos report --batch --tmp-dir $TMT_PLAN_DATA
```

Adding to the backlog as a nice-to-have.
